### ISSUE TITLE
Regarding the citation of paper data?

### ISSUE THREAD

**Issue by apuomline** [sotiraslab/AgileFormer#15]
Hello, author. First of all, thank you for your work, which is very inspiring to us. However, we found that the effect of your work on the Synapse dataset is consistent with that in HiFormer. In other words, can we directly cite the data of other papers that have run the Synapse dataset? I am a beginner and I don't quite understand this aspect. Looking forward to your answer! Translate this content into English.
![联想截图_20240813140948](https://github.com/user-attachments/assets/0d7bca7b-cffc-4821-b1b0-3bc6096c0d58)

**Comment by peijieq** [comment]
Thanks for being interested in our work. Could you elaborate on your question a bit more ? I guess you were trying to say if it's fine to cite the performance from the other papers (e.g., HiFormer, TransUNet, SwinUNet) on Synapse dataset. My answer is yes, as long as you are following the same experimental protocol as those papers (or simply building upon their code), it is totally reasonable. We cited their empirical results directly since we followed the same experimental protocols. If pre-trained weights are released by the authors (e.g., our AgileFormer, nnFormer, etc), you are definitely welcome to test them yourself. Hope this will answer your question.

**Comment by apuomline** [comment]
OK, thanks for your reply. We have two more questions for you. Question 1: Can you explain specifically what the same experimental protocol refers to? Is it the same experimental configuration? For example, are the settings for lr, weight_decay, batch_size, etc.? Question 2: You said that you can directly quote the experimental results of the SwinUNet paper on Synapse. In other words, you do not need to retrain SwinUNet on Synapse. But I have seen that the author team of SwinUNet did not open source their pre-trained weights on github. So how did you get the segmentation result map of SwinUNet on the Synapse data set? This question has troubled us for a long time, and we look forward to your answer.
![image](https://github.com/user-attachments/assets/0079fe03-84a4-484e-8d3c-b1c9e6ef6da9)

**Comment by peijieq** [comment]
To your first question, What I meant by the same experimental protocol was mainly the training/testing split, preprocessing, and image size; whereas hyperparameters (lr, weight_decay, num of epochs, batch size, etc) can vary from model to model (i.e., what we typically refer to as hyperparameter tuning). To your second question, yes. We retrained the SwinUNet to reproduce their results (79.13) for visualization. You can contact me at my email: peijie.qiu@wustl.edu if you want some results for visualization. I also shared the visualization code in this repo.

**Comment by apuomline** [comment]
Thank you very much for your reply.However, your model hyperparameters, such as lr, batch_size, etc., are not consistent with swinunet's hyperparameters. , how can you possibly reproduce the segmentation results consistent with swinunet? (Even if the same machine, the same hyperparameter configuration, is retrained and tested twice, the segmentation results of the two times cannot be the same, right?) Or, the segmentation result you reproduce is not 79.13, but you will log in locally The segmentation results of swinunet are visualized. That is, the segmentation score corresponding to the swinunet segmentation result map in your paper is not 79.13?

**Comment by apuomline** [comment]
I don't mean any harm. I never understood how to directly quote data from other papers, but I couldn't get the corresponding segmentation result map. The data in the paper cannot be reproduced locally. We also thought of this approach, but we want to confirm whether this approach is reasonable.

**Comment by peijieq** [comment]
I want to make it very clear. What I meant by reproducing SwinUnet’s results was done by its code and certainly what they identified as the optimal hyperparameters, instead of the configurations in our settings. Indeed, we can reproduce SwinUnet’s results as the authors have the random seed fixed (I tried this on different machines, the results were the same). You’ll expect to produce the same results even on different machines, though minor deviations may be possible due to numerical issues of your system. For the other hyperparameters, it can definitely vary from model to model: even when you change a small portion of the model, your “optimal” hyperparameters should be reset. With that said, I think the most convincing way is to open source the segmentation results, which I will do shortly. But I cannot guarantee for other models.
